client/components/data/query-products-list/index.ts

```diff
@@ -0,0 +1,57 @@
+import { useEffect } from 'react';
+import { useDispatch } from 'react-redux';
+import { requestProductsList } from 'calypso/state/products-list/actions';
+import { isProductsListFetching, getProductsListType } from 'calypso/state/products-list/selectors';
+import type { CalypsoDispatch } from 'calypso/state/types';
+import type { AppState } from 'calypso/types';
+
+type QueryProductsListProps = {
+	type?: string;
+	currency?: string;
+	persist?: boolean;
+	productSlugList?: string[];
+};
+
+type RequestProps = Omit< QueryProductsListProps, 'productSlugList' > & {
+	product_slugs?: string[];
+};
+
+const request =
+	( { persist, ...props }: RequestProps ) =>
+	( dispatch: CalypsoDispatch, getState: () => AppState ) => {
+		if (
+			isProductsListFetching( getState() ) ||
+			( persist && props.type === getProductsListType( getState() ) )
+		) {
+			return;
+		}
+
+		dispatch( requestProductsList( props as any ) );
+	};
+
+export function useQueryProductsList( {
+	type = 'all',
+	currency,
+	persist,
+	productSlugList,
+}: QueryProductsListProps = {} ) {
+	const dispatch = useDispatch();
+
+	// Only runs on mount.
+	useEffect( () => {
+		// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+		// @ts-ignore
+		dispatch( request( { type, currency, persist, product_slugs: productSlugList } ) );
+	}, [ dispatch, type, persist, currency, productSlugList ] );
```

Comment:
`productSlugList` is an array, and it can be different on every render. There are usages like this:
```js
<QueryProductsList productSlugList={ [ 'value_bundle', 'personal-bundle', 'business-bundle' ] } />
```
We'd be better off if we used a string instead of an array:
```js
<QueryProductsList productSlugList="value_bundle,personal-bundle,business-bundle" />
```
This `,`-delimited string will be the value of the `product_slugs` query param.

Reply:
Searching the code, I couldn't find any instance of `<QueryProductsList />` passing `productSlugList`, just found one for the hook counterpart. This can be something refactored together with the hidden `site` param in a follow-up PR.